pkg/controller/service/servicecontroller.go

```diff
@@ -125,16 +127,33 @@ func New(
 		nodeLister:       nodeInformer.Lister(),
 		nodeListerSynced: nodeInformer.Informer().HasSynced,
 		workingQueue:     workqueue.NewNamedDelayingQueue("service"),
+		endpoints: make(map[string]int),
 	}
 
 	serviceInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: s.enqueueService,
 			UpdateFunc: func(old, cur interface{}) {
+				endpointsChanged := false
 				oldSvc, ok1 := old.(*v1.Service)
 				curSvc, ok2 := cur.(*v1.Service)
+				oldEndpoints := s.endpoints[oldSvc.ObjectMeta.Name]
+				newEndpoints := s.getNumberOfEndpoints(oldSvc)
+				if oldEndpoints != newEndpoints {
+					endpointsChanged = true
+				s.endpoints[oldSvc.ObjectMeta.Name] = newEndpoints
+				}
 				if ok1 && ok2 && s.needsUpdate(oldSvc, curSvc) {
 					s.enqueueService(cur)
+				} else if wantsLoadBalancer(curSvc) && endpointsChanged {
+					nodes, err := s.nodeLister.List(labels.Everything())
+					err = s.balancer.UpdateLoadBalancer(s.clusterName, curSvc, nodes)
+					if err == nil {
+						s.eventRecorder.Event(curSvc,
+							api.EventTypeNormal, 
+							"UpdatedLoadBalancer", 
+							"Updated load balancer with new Endpoints")
```

Comment:
Can we handle event description in more generic way to handle 'add' and 'delete' members? This gives me an impression that it is considering addition of new endpoints only.